### PR TITLE
[NFC] Add RTL README

### PR DESCRIPTION
**Pull request by sebastiankreutzer**
## Description
Adds a README to the RTL to document basic API usage and runtime configuration. 

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor / internal change (no functional change)
- [x] Documentation update

## Quality Checklist
*Please ensure the following are completed before requesting review:*

* [ ] Runtime-compatible (no externally visible existing method has its parameters changed, was deleted or has its exception behavior changed)
* [ ] Compile-time compatible (new arguments & properties added have default values, but users need to recompile against the new version to work)

IR:
* [ ] Introduces new types into the SDFG IR
* [ ]  Changed / added properties on SDFG IR Elements (LibraryNodes, ControlFlowNodes DataFlowNodes etc.)
  *  [ ] New / modified types have serialization / deserialization support
  *  [ ] Deserialization is backward compatible (new serialization can deserialize old format and represent it in the new format if applicable)
  *  [ ] Serialized format is backward compatible (previous deserializer can read new serialized format without incorrectness (just lacking additional guarantees that are optional or did not exist before)
  *  [ ] Serializer has an option to emit the previous format

### Unit Tests
- [ ] New unit tests have been added for the changes.
- [ ] Existing unit tests pass locally.
- [ ] Edge cases and error paths are covered.

### Integration Tests
- [ ] New integration tests have been added (or existing ones updated).
- [ ] Integration tests pass locally.
- [ ] Interactions with other components/services are verified.

### Documentation
- [ ] Public APIs, options, and behavior changes are documented.
- [x] README / docs site / inline comments updated where applicable.
- [ ] Changelog / release notes updated (if applicable).

### Test Coverage
- [ ] Test coverage has not decreased as a result of this change.
- [ ] Coverage report reviewed and acceptable.
- [ ] Critical / new code paths are covered.

### CODE REVIEW

File: rtl/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+### Configuration
+
+Tracing can be configured using the following environment variables:
+- `DAISY_PAPI_PATH`: Path to the PAPI library. System paths are used if not specified.
+- `__DAISY_PAPI_VERSION` (required): PAPI library version (e.g., `0x07020000` for version 7.2.0.0)
+- `__DAISY_INSTRUMENTATION_MODE`: Set to `aggregate` to activate aggregate mode (default is full tracing)
+- `__DAISY_INSTRUMENTATION_FILE`: Output location of the trace file
+- `__DAISY_INSTRUMENTATION_EVENTS`: List of PAPI events to record on the CPU
+- `__DAISY_INSTRUMENTATION_EVENTS_CUDA`: List of PAPI events to record on the GPU
+- `__DAISY_INSTRUMENTATION_ASYNC_CAPTURE` Enable/disable asynchronous event capturing (on by default)
```
Comment (ramonwirsch) [inline rtl/README.md]: Async capture is on by default for aggregate + GPU EVENT Set with CUDA or ROCM

File: rtl/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+For GPU kernels, use `__DAISY_EVENT_SET_CUDA` instead (works for ROCm too). Counters are disabled using `__DAISY_EVENT_SET_NONE`.
+
+Regions are subsequently started using the unique region ID:
+
+```
+__daisy_instrumentation_enter(region_id);
+// Work happens here...
+__daisy_instrumentation_exit(region_id);
+```
+
```
Comment (ramonwirsch) [inline rtl/README.md:35]: we use the space between exit and finalize to record "static" counters, that the compiler can compute. Like bytes transferred or flops.

File: rtl/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Call `__daisy_instrumentation_finalize(region_id)` after the final measurement to finish recording. 
+
+For the full list of API calls and metadata fields, refer to [rtl/include/daisy_rtl/daisy_rtl.h](include/daisy_rtl/daisy_rtl.h).
+
```
Comment (ramonwirsch) [inline rtl/README.md:42]: finalize_all() is required, otherwise the async capture would not count any times.

**Review by ramonwirsch — CHANGES_REQUESTED** [review]

**Review by ramonwirsch — APPROVED** [review]
